samples/snippets/src/main/java/com/example/spanner/TracingSample.java

```diff
@@ -87,6 +97,9 @@ public static void main(String[] args) throws Exception {
     } finally {
       // Closes the client which will free up the resources used
       spanner.close();
+      StackdriverExporter.unregister();
+      StackdriverStatsExporter.unregister();
+      System.exit(0);
```

Comment:
This seems overly harsh. What is the reason that we need to call `System.exit(0)` to stop the application here? Is the Stackdriver exporter using non-daemon threads that are not stopped? If so, does the `StackdriverExporter` have a close/shutdown method?

Reply:
StackdriverExporter does not export any close/shutdown method they only have unregister, without system.exit I see exceptions and sample getting stuck https://pastebin.com/drh6qubH

Reply:
The 'getting stuck' problem is actually caused by `ZPageHandlers.startHttpServerAndRegisterAll(8080);`. That starts an HTTP server that can be accessed on `http://localhost:8080/tracez` to view the traces that were collected. That component does not have a public `stop()` method, and keeps the JVM alive, as it uses non-daemon threads. Can we update the comment here to reflect this?

The errors regarding closed channels could in theory still happen. I think that there is a race condition in the OpenCensus stats exporter that tries to flush any remaining stats, while at the same time also closing the underlying exporter client. Those can be ignored.